web/packages/teleport/src/Navigation/SideNavigation/Navigation.tsx

```diff
@@ -86,6 +88,20 @@ export type NavigationSubsection = {
   icon: (props) => JSX.Element;
   parent?: TeleportFeature;
   searchableTags?: string[];
+  /**
+   * customRouteMatchFn is a custom function for determining whether this subsection is currently active,
+   * this is useful in cases where a simple base route match isn't sufficient.
+   */
+  customRouteMatchFn?: (currentViewRoute: string) => boolean;
+  /**
+   * subCategory is the subcategory (ie. subsection grouping) this subsection should be under, if applicable.
+   * */
+  subCategory?: CustomNavigationSubcategory;
```

Comment:
I'd avoid adding very specific properties to the general API. `subCategory` makes sense only in context of `ResourcesSection`, for any other section it does nothing. This makes the API harder to maintain since it is not clear how to use given the property.
Instead, I'd extend `NavigationSection` with a type that has `resourceView: 'all' | 'filtered'`, something like `ResourcesNavigationSection `.